addon-test-support/index.js

```diff
@@ -0,0 +1,57 @@
+import { get } from '@ember/object';
+import { getContext, settled } from '@ember/test-helpers';
+import Test from 'ember-simple-auth/authenticators/test';
+
+const SESSION_SERVICE_KEY = 'service:session';
+const TEST_CONTAINER_KEY = 'authenticator:test';
+
+function ensureAuthenticator(owner) {
+  const authenticator = owner.lookup(TEST_CONTAINER_KEY);
+  if (!authenticator) {
+    owner.register(TEST_CONTAINER_KEY, Test);
+  }
+}
+
+/**
+ * Authenticates the session.
+ *
+ * @param {Object} sessionData Optional argument used to mock an authenticator
+ * response (e.g. a token or user).
+ * @return {Promise}
+ * @public
+ */
+export function authenticateSession(sessionData) {
+  const { owner } = getContext();
+  const session = owner.lookup(SESSION_SERVICE_KEY);
+  owner.setupRouter(); // router must initialize fully before authentication
+  ensureAuthenticator(owner);
+  session.authenticate(TEST_CONTAINER_KEY, sessionData);
+  return settled();
```

Comment:
Is authenticating the session an asynchronous task? 
If it's not, you don't need to return `settled()`, and if it is I would make `authenticateSession` an async function. I think it's sync.

Reply:
`session.authenticate` returns a promise, and an internal call there events are fired that work async (here https://github.com/simplabs/ember-simple-auth/blob/master/addon/internal-session.js#L117) then another call to `_callStoreAsync`, so looks pretty async.

Reply:
Indeed. In that case I think the best approach is to make the function async:
```js
export async function authenticateSession(sessionData) {
  const { owner } = getContext();
  const session = owner.lookup(SESSION_SERVICE_KEY);
  owner.setupRouter();
  ensureAuthenticator(owner);
  await session.authenticate(TEST_CONTAINER_KEY, sessionData);
  return settled();
}
```